src/glossary.adoc

```diff
@@ -0,0 +1,116 @@
+[[sec:glossary]]
+[glossary]
+= Glossary
+
+[glossary]
+
+arity::
+The number of arguments declared by or passed to a macro.  A macro can be _fixed_ or _variable_
+arity; the number of subforms in an invocation is its _actual arity_.
+
+argument::
+A single subform within an encoding expression, corresponding to one of the macro’s parameters.
+This is a one-to-one relation except in the case of rest parameters, where any number of trailing
+arguments are passed to the same parameter.
+
+cardinality::
+The number of elements accepted by a single argument of a macro, or produced by a macro. The
+arguments and results of a macro are streams, and each can be constrained to have exactly-one
+(`*!*`), zero-or-one (`*?*`), zero-or-more (`*{asterisk}*`, `*\...*`), or one-or-more (`*+*`)
+elements.
```

Comment:
I think this is _argument cardinality_ and we also have _stream cardinality_ which is the number of values that are associated with a parameter (really any evaluation).